core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -38,7 +38,14 @@
 .section .data
 .balign 4
 
-#ifdef CFG_BOOT_SYNC_CPU
+/*
+ * For VxWorks Support, the primary core will go through OP-TEE init
+ * without ever touching secondary cores; it is only after the primary
+ * core enters VxWorks, it issues special SMC to request booting of
+ * secondary cores. Thus CPU Sync is not required.
+ */
+
+#if defined (CFG_BOOT_SYNC_CPU) && !defined (CFG_TEE_VXWORKS_SUPPORT)
```

Comment:
> We want to reuse the existing plat-ls to support VxWorks

i would rather propose plat-stm and plat-ls to be changed:
-$(call force,CFG_BOOT_SYNC_CPU,y)
+CFG_BOOT_SYNC_CPU ?= y

To run VxWorks on a native optee/ls:
make CFG_BOOT_SYNC_CPU=n

Otherwise, in mk/config.mk
+ifeq ($(CFG_TEE_VXWORKS_SUPPORT),y)
+$(call force,CFG_BOOT_SYNC_CPU,n)
+endif

would that work?

My point is you intend to disable the full "boot sync cpu". So i would
rather have it disabled than adding extra check in the source code.


> It seems the generic boot process implementation is targeted for Firmware
Code.
> The firmware would bring up all cores and put each through the OP-TEE;
>  VxWorks's usage differs from this:

no so. ATF support on aarch32 expects configuration sets explicitly
CFG_WITH_ARM_TRUSTED_FW=y
CFG_BOOT_SYNC_CPU=n

CFG_BOOT_SYNC_CPU is designed for that purpose.
I think the real issue is plat-ls (as plat-stm) forcing cpu sync at boot.


On 12 August 2016 at 09:51, yanyan-wrs <notifications@github.com> wrote:

> In core/arch/arm/kernel/generic_entry_a32.S
> <https://github.com/OP-TEE/optee_os/pull/936#discussion_r74552988>:
>
> > @@ -38,7 +38,14 @@
> >  .section .data
> >  .balign 4
> >
> > -#ifdef CFG_BOOT_SYNC_CPU
> > +/*
> > + * For VxWorks Support, the primary core will go through OP-TEE init
> > + * without ever touching secondary cores; it is only after the primary
> > + * core enters VxWorks, it issues special SMC to request booting of
> > + * secondary cores. Thus CPU Sync is not required.
> > + */
> > +
> > +#if defined (CFG_BOOT_SYNC_CPU) && !defined (CFG_TEE_VXWORKS_SUPPORT)
>
> I understand we can set CFG_BOOT_SYNC_CPU=n in plat-xxx/conf.mk. However,
> the intention here is to "uniformly" disable the CPU sync feature if OP-TEE
> is to be used together with VxWorks. Besides, for some platform, for
> example, plat-ls, it already defines the CFG_BOOT_SYNC_CPU=y. We want to
> reuse the existing plat-ls to support VxWorks, but we cannot just rudely
> overwrite CFG_BOOT_SYNC_CPU=n, so we have to disable the feature using the
> CFG_TEE_VXWORKS_SUPPORT flag.
>
> It seems the generic boot process implementation is targeted for Firmware
> Code. The firmware would bring up all cores and put each through the
> OP-TEE; VxWorks's usage differs from this: only the primary core is
> required to go through the OP-TEE and enters NS OS (this maximize
> compatibility as it is not required to have TrustZone aware firmware), then
> NS OS asks the OP-TEE to bring up the rest cores, so there's no need to
> sync between primary and secondary cores.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/OP-TEE/optee_os/pull/936/files/94967ff72e6c157bd08e0134b1e82a5792cfd333#r74552988>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATEHJt963UI4PbeVfCiQwQ5bnEweMhw1ks5qfCYBgaJpZM4JTw0v>
> .
>